transport/api/src/adaptors/ping.rs

```diff
@@ -60,7 +59,7 @@ impl<R: PeerAddressResolver + std::marker::Sync> PingExternalAPI for PingExterna
                 status.get_quality(),
                 status.get_average_quality()
             );
-            if let Ok(pk) = OffchainPublicKey::from_peerid(peer) {
+            if let Ok(pk) = OffchainPublicKey::try_from(*peer) {
```

Comment:
Would it be possible to do a `try_from` override for the `&PeerId` type as well? To avoid copies...